src/libp2p/connectionsManager.ts

```diff
@@ -283,6 +284,41 @@ export class ConnectionsManager {
     return certRegister
   }
 
+  public registerUserCertificate = async (userCsr: UserCsr) => {
+    const response = await this.sendCertificateRegistrationRequest(userCsr)
+    if (response.status !== 200) {
+      switch (response.status) {
+        case 403:
+          break
+      }
+      return
+    }
+    const certificate: string = await response.json()
+
+    await this.saveCertificate(certificate)
+
+    this.io.emit(EventTypesResponse.SEND_USER_CERTIFICATE, certificate)
+  }
+
+  public sendCertificateRegistrationRequest = async (userCsr: UserCsr): Promise<Response> => {
+    const options = {
+      method: 'POST',
+      body: JSON.stringify({ data: userCsr }),
+      headers: { 'Content-Type': 'application/json' },
+      agent: () => {
+        return new SocksProxyAgent({ port: this.agentPort, host: this.agentHost })
+      }
+    }
+    try {
+      return await fetchAbsolute(fetch)(
+        'http://wzispgrbrrkt3bari4kljpqz2j6ozzu3vlsoi2wqupgu7ewi4ncibrid.onion:7789'
```

Comment:
this should be parameter

Reply:
But parameter that is passed with websocket or should it be some environment variable? Emilia says that this address is fixed

Reply:
This address is fixed temporarily, because we are not developing everything at once.

Eventually this address will be part of "invitation to zbay community" so will come from frontend. It should be passed with websocket